src/main/java/com/openshift/internal/restclient/model/DeploymentConfig.java

```diff
@@ -164,4 +188,79 @@ public IDeploymentTrigger addTrigger(String type) {
 	public String getDeploymentStrategyType() {
 		return asString(DEPLOYMENTCONFIG_STRATEGY);
 	}
+
+	@Override
+	public boolean haveTriggersFired() {
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		if (causes == null || causes.getType() != ModelType.LIST)
+			return false;
+		return causes.asList().size() > 0;
+	}
+
+	@Override
+	public boolean didImageTrigger(String imageTag) {
+		if (!haveTriggersFired())
+			return false;
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		for (ModelNode cause : causes.asList()) {
+			String type = cause.get(TYPE).asString();
```

Comment:
The base class has 'asString(ModelNode,key)' to do this transformation.  Additionally the 'asXXXXX' methods return defaults instead of 'undefined' nodes